src/java/picard/analysis/CollectWgsMetrics.java

```diff
@@ -349,12 +358,18 @@ protected WgsMetrics generateWgsMetrics() {
         return new WgsMetrics();
     }
 
+    /**
+     * By design, if an INTERVALS file is provided, we want to count just those bases at the positions we care about,
+     * not across the entire read. Therefore, we call filter.getFilteredRecords() so that only the bases in the pileup
+     * at a given position are included in the calculations (with filter.getFilteredBases() we would be including other
+     * bases in the read too).
+     */
     protected long getBasesExcludedBy(final CountingFilter filter) {
-        return filter.getFilteredBases();
+        return (INTERVALS != null) ? filter.getFilteredRecords() : filter.getFilteredBases();
```

Comment:
but they will be! we want to use this to filter on 1-22,x,y,mt...right?

On Wed, Mar 2, 2016 at 4:58 PM, Eric Banks <notifications@github.com> wrote:

> In src/java/picard/analysis/CollectWgsMetrics.java
> <https://github.com/broadinstitute/picard/pull/470#discussion_r54799033>:
>
> >      protected long getBasesExcludedBy(final CountingFilter filter) {
> > -        return filter.getFilteredBases();
> > +        return (INTERVALS != null) ? filter.getFilteredRecords() : filter.getFilteredBases();
>
> Yeah, that's hard. That's why it was called "FromSampledSites".
> @yfarjoun <https://github.com/yfarjoun> what if we check up front and
> fail if any of the intervals are greater than 1 base?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/broadinstitute/picard/pull/470/files#r54799033>.
>